.github/workflows/builder_node_slsa3.yml

```diff
@@ -0,0 +1,388 @@
+# Copyright The GOSST team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: SLSA node builder
+
+permissions:
+  contents: read
+
+env:
+  # Builder.
+  BUILDER_BINARY: slsa-builder-go-linux-amd64 # Name of the binary in the release assets.
+  BUILDER_DIR: internal/builders/node # Source directory if we compile the builder.
+
+defaults:
+  run:
+    shell: bash
+
+###################################################################
+#                                                                 #
+#            Input and output argument definitions                #
+#                                                                 #
+###################################################################
+on:
+  workflow_call:
+    secrets:
+      token:
+        description: >
+          Optional token.
+
+          This argument is passed, unchanged, to `actions/node-setup`'s
+          as `token` parameter.
+        required: false
+        # Note: set to the same value as actions/node-setup https://github.com/actions/setup-node/blob/main/action.yml#L19.
+        default: ${{ github.token }}
+    inputs:
+      node-version:
+        description: >
+          The node version to use.
+
+          This argument is passed, unchanged, to `actions/node-setup`'s
+          as `node-version` parameter.
+        required: true
+        type: string
+      # TODO: support uploading assets to GitHub release.
```

Comment:
nit: Consider adding a github issue and adding the URL to the comment. Same for other TODOs.

Reply:
We should also add a new milestone and add these issues to it.

Reply:
I added a [milestone](https://github.com/slsa-framework/slsa-github-generator/milestone/8) that we can add future issues to and can track publicly as part of the roadmap.

Reply:
yeah I was planning on creating an issue for it. Thanks for the reminder.